lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -83,14 +84,17 @@ def load_from_data(self, model, course_structure, scores_client, submissions_sco
         """
         for block in model.visible_blocks.blocks:
             persisted_values = {'weight': block.weight, 'possible': block.max_score}
+            locator = BlockUsageLocator.from_string(block.locator)
             self._compute_block_score(
                 User.objects.get(id=model.user_id),
-                block.locator,
+                locator,
                 course_structure,
                 scores_client,
                 submissions_scores,
                 persisted_values
             )
+        # self.scores may hold outdated data, force it to refresh on next access
+        lazy.invalidate(self, 'scores')
```

Comment:
This is actually unneeded - removing this line doesn't help or hurt us in any way. I believe the problem outlined in https://github.com/edx/edx-platform/pull/13324 is at the root of the "grade not updating" behavior I was seeing.